torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -0,0 +1,472 @@
+import copy
+import re
+from typing import List, Optional
+
+import torch
+
+# import cutlass libs
+import gemm_operation as cutlass_gemm_op
+import library as cutlass_lib
+
+from ...ir import Buffer, FixedLayout, IRNode, Layout
+from ..common import IndentedBuffer
+
+from . import cutlass_utils
+from .cuda_kernel import CUDATemplateKernel
+from .cuda_template import CutlassTemplate
+
+# Only supports alpha * A@B + beta * C now.
+# TODO: Support arbitrary epilogue after epilogue visitor is released in cutlass 3.2.
+GEMM_TEMPLATE = r"""
+{{template.header().getvalue()}}
+{{template.globals().getvalue()}}
+{{instance_definition}}
+// When workspace_size is not a nullptr, populates requested workspace_size and returns.
+// Otherwise, compuates the Gemm kernel using the given workspace ptr.
+extern "C" {
+{{kernel.def_kernel(inputs=[X, W, Bias], outputs=[Y], names_str="X, W, Bias, Y", input_reorder=input_reorder)}} {
+  try {
+  // printf("X: %p\n", (void*)(X));
```

Comment:
We could conditionally generate these in if (in an uncommented manner) if some debug flag is set, otherwise I think they should not become part of the generated code.

Reply:
Yes didn't fix it in draft code ^^.